libs/beacon/src/beacon_service.cpp

```diff
@@ -334,26 +392,61 @@ BeaconService::State BeaconService::OnVerifySignaturesState()
  */
 BeaconService::State BeaconService::OnCompleteState()
 {
+  beacon_state_gauge_->set(static_cast<uint64_t>(state_machine_->state()));
   FETCH_LOCK(mutex_);
 
   uint64_t const index = block_entropy_being_created_->block_number;
-  beacon_entropy_last_generated_->set(index);
-  beacon_entropy_generated_total_->add(1);
 
-  // Populate the block entropy structure appropriately
-  block_entropy_being_created_->group_signature =
-      active_exe_unit_->manager.GroupSignature().getStr();
+  if (completed_block_entropy_.find(index) == completed_block_entropy_.end())
```

Comment:
This should never happen (that entropy has already been created when you get to this state)